File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/ExceptionEntry.java
package pt.sapo.gis.trace;

import java.util.Map;
import pt.sapo.gis.trace.configuration.SeverityType;

/**
 * @author GIS Team
 * Class for entry with an exception
 */
public class ExceptionEntry extends Entry{
    public static final String EXCEPTION_PROPERTY_NAME = "exception";
            
    public Exception getException() {
        return ((ExceptionEntryProperty) getProperty(EXCEPTION_PROPERTY_NAME)).getException();
    }

    public final void setException(Exception e) {                
        setProperty(EXCEPTION_PROPERTY_NAME,  new ExceptionEntryProperty(e));
    }
    
    public ExceptionEntry(){}
    
    public ExceptionEntry(String description, SeverityType severity, Exception exception){    
        super(description, severity);
        setException(exception);
    }
    
    public ExceptionEntry(String description, SeverityType severity, Map<String,Object> properties, Exception exception){    
        super(description, severity, properties);
        setException(exception);
    }
}

File: SMI-SDK/.NET SDK/Library/gis/trace/ExceptionEntry.cs
namespace pt.sapo.gis.trace
{
    using System;
    using pt.sapo.gis.trace.configuration;
    using System.Collections.Generic;

    /**
     * @author GIS Team
     * Class for entry with an exception
     */
    public class ExceptionEntry : Entry
    {
        public static readonly String EXCEPTION_PROPERTY = "exception";

        public Exception Exception
        {
            get { return ((ExceptionEntryProperty)Properties[EXCEPTION_PROPERTY]).Exception; }
            protected set { Properties.Add(EXCEPTION_PROPERTY, new ExceptionEntryProperty(value)); }
        }

        public ExceptionEntry() { }

        public ExceptionEntry(String description, severityType severity, Exception exception) : base(description, severity) {
            Exception = exception;
        }

        public ExceptionEntry(String description, severityType severity, IDictionary<String, Object> properties, Exception exception) : base(description, severity, properties) {
            Exception = exception;
        }
    }
}